### DISCUSSION TITLE
Relay the HLS stream to a HTTP server via post

### DISCUSSION THREAD

**Discussion by itprotj** [AzuraCast/AzuraCast#6258] [Q&A]
Hi all,

Not sure if this is even possible however I'll try to describe my current situation and hopeful outcome:

Network 1 is based in a datacenter and has little to no restrictions incoming/outgoing wise with ports. Azuracast runs on this network.

Network 2 is behind a heavily filtered firewall meaning to access Network 1 I would need authentication each time I goto an IP/website in Network 1. I have a server on this network that I am able to POST data to from the outside which allows me to work with these restrictions. I am only able to work with ports 80/443 so I am restricted to a web based way of getting the azuracast stream in.

My desired outcome (if possible and please suggest things to if you can!), for Azuracast on network 1 to HTTP post the HLS stream to the web server on network 2. Network 2 then receives the data as posts and saves the HLS/TS segments as required. I could then load the HLS stream within network two from that web server.

Hope someone can suggest ways to accomplish this.

Thanks

**Comment by BusterNeece** [comment]
We have to do some similar juggling in some projects I've worked on, and I've seen folks using a little open-source tool called Restreamer to accomplish this: https://docs.datarhei.com/restreamer/

It can ingest just about any format and output just about any other, while also serving as a direct relay in situations like this.

May be worth a look :)

**Reply by itprotj** [reply]
Thanks buddy, I'll take a look at it.. Looks promising!
